### ISSUE TITLE
Its Not Working

### ISSUE THREAD

**Issue by BTaskaya** [bonnici/node-steam-chat-bot#11]
When I Try Node Example.JS
`node example.js
module.js:471
    throw err;
    ^

Error: Cannot find module 'steam-chat-bot'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Users\CezmiShef\a\t\example.js:1:102)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)`

**Comment by bonnici** [comment]
You need to run `npm install` and get that to succeed if you haven't already.

**Comment by RealRodneyMcKay** [comment]
i runned it. 
```
`C:\Users\CezmiShef>cd a

C:\Users\CezmiShef\a>cd t

C:\Users\CezmiShef\a\t>npm install
npm WARN steam-chat-bot@1.2.3 No license field.

C:\Users\CezmiShef\a\t>dir
 Volume in drive C has no label.
 Volume Serial Number is E820-2691

 Directory of C:\Users\CezmiShef\a\t

27-May-17  04:13 PM    <DIR>          .
27-May-17  04:13 PM    <DIR>          ..
27-May-17  04:11 PM             5,236 example.js
22-Feb-16  02:06 AM    <DIR>          lib
28-May-17  01:38 PM    <DIR>          node_modules
22-Feb-16  02:06 AM               819 package.json
22-Feb-16  02:06 AM    <DIR>          spec
               2 File(s)          6,055 bytes
               5 Dir(s)  56,300,015,616 bytes free

C:\Users\CezmiShef\a\t>cd node_modules

C:\Users\CezmiShef\a\t\node_modules>dir
 Volume in drive C has no label.
 Volume Serial Number is E820-2691

 Directory of C:\Users\CezmiShef\a\t\node_modules

28-May-17  01:38 PM    <DIR>          .
28-May-17  01:38 PM    <DIR>          ..
27-May-17  04:12 PM    <DIR>          .bin
27-May-17  04:12 PM    <DIR>          @protobufjs
27-May-17  04:12 PM    <DIR>          @types
27-May-17  04:11 PM    <DIR>          abbrev
27-May-17  04:11 PM    <DIR>          adm-zip
27-May-17  04:11 PM    <DIR>          ajv
27-May-17  04:11 PM    <DIR>          ansi-regex
27-May-17  04:11 PM    <DIR>          ansi-styles
27-May-17  04:11 PM    <DIR>          aproba
27-May-17  04:11 PM    <DIR>          are-we-there-yet
27-May-17  04:11 PM    <DIR>          asn1
27-May-17  04:11 PM    <DIR>          assert-plus
27-May-17  04:11 PM    <DIR>          assertion-error
27-May-17  04:12 PM    <DIR>          async
27-May-17  04:11 PM    <DIR>          asynckit
27-May-17  04:11 PM    <DIR>          aws-sign2
27-May-17  04:11 PM    <DIR>          aws4
27-May-17  04:11 PM    <DIR>          balanced-match
27-May-17  04:11 PM    <DIR>          bcrypt-pbkdf
27-May-17  04:11 PM    <DIR>          bindings
27-May-17  04:12 PM    <DIR>          bl
27-May-17  04:11 PM    <DIR>          block-stream
27-May-17  04:11 PM    <DIR>          boom
27-May-17  04:11 PM    <DIR>          brace-expansion
27-May-17  04:11 PM    <DIR>          buffer-crc32
27-May-17  04:11 PM    <DIR>          buffer-shims
27-May-17  04:11 PM    <DIR>          bytebuffer
27-May-17  04:11 PM    <DIR>          capture-stack-trace
27-May-17  04:11 PM    <DIR>          caseless
27-May-17  04:11 PM    <DIR>          chai
27-May-17  04:11 PM    <DIR>          chalk
27-May-17  04:11 PM    <DIR>          cleverbot-node
27-May-17  04:11 PM    <DIR>          co
27-May-17  04:11 PM    <DIR>          code-point-at
27-May-17  04:11 PM    <DIR>          colors
27-May-17  04:11 PM    <DIR>          combined-stream
27-May-17  04:11 PM    <DIR>          commander
27-May-17  04:11 PM    <DIR>          concat-map
27-May-17  04:11 PM    <DIR>          console-control-strings
27-May-17  04:11 PM    <DIR>          core-util-is
27-May-17  04:11 PM    <DIR>          create-error-class
27-May-17  04:11 PM    <DIR>          cryptiles
27-May-17  04:11 PM    <DIR>          crypto
27-May-17  04:11 PM    <DIR>          ctype
27-May-17  04:11 PM    <DIR>          cycle
27-May-17  04:12 PM    <DIR>          dashdash
27-May-17  04:11 PM    <DIR>          debug
27-May-17  04:11 PM    <DIR>          deep-eql
27-May-17  04:11 PM    <DIR>          deep-equal
27-May-17  04:11 PM    <DIR>          deep-extend
27-May-17  04:11 PM    <DIR>          delayed-stream
27-May-17  04:11 PM    <DIR>          delegates
27-May-17  04:11 PM    <DIR>          duplexer2
27-May-17  04:11 PM    <DIR>          ecc-jsbn
27-May-17  04:11 PM    <DIR>          error-ex
27-May-17  04:11 PM    <DIR>          escape-string-regexp
27-May-17  04:11 PM    <DIR>          extend
27-May-17  04:11 PM    <DIR>          extsprintf
27-May-17  04:11 PM    <DIR>          eyes
27-May-17  04:11 PM    <DIR>          forever-agent
27-May-17  04:11 PM    <DIR>          form-data
27-May-17  04:11 PM    <DIR>          formatio
27-May-17  04:11 PM    <DIR>          fs.realpath
27-May-17  04:11 PM    <DIR>          fstream
27-May-17  04:11 PM    <DIR>          fstream-ignore
27-May-17  04:11 PM    <DIR>          gauge
27-May-17  04:11 PM    <DIR>          generate-function
27-May-17  04:11 PM    <DIR>          generate-object-property
27-May-17  04:12 PM    <DIR>          getpass
27-May-17  04:11 PM    <DIR>          glob
27-May-17  04:11 PM    <DIR>          google-images
27-May-17  04:11 PM    <DIR>          got
27-May-17  04:11 PM    <DIR>          graceful-fs
27-May-17  04:11 PM    <DIR>          graceful-readlink
27-May-17  04:11 PM    <DIR>          har-schema
27-May-17  04:11 PM    <DIR>          har-validator
27-May-17  04:11 PM    <DIR>          has-ansi
27-May-17  04:11 PM    <DIR>          has-unicode
27-May-17  04:11 PM    <DIR>          hawk
27-May-17  04:11 PM    <DIR>          hoek
27-May-17  04:11 PM    <DIR>          http
27-May-17  04:11 PM    <DIR>          http-signature
27-May-17  04:11 PM    <DIR>          i
27-May-17  04:11 PM    <DIR>          inflight
27-May-17  04:11 PM    <DIR>          inherits
27-May-17  04:11 PM    <DIR>          ini
27-May-17  04:11 PM    <DIR>          is-arrayish
27-May-17  04:11 PM    <DIR>          is-fullwidth-code-point
27-May-17  04:11 PM    <DIR>          is-my-json-valid
27-May-17  04:11 PM    <DIR>          is-property
27-May-17  04:11 PM    <DIR>          is-redirect
27-May-17  04:11 PM    <DIR>          is-retry-allowed
27-May-17  04:11 PM    <DIR>          is-stream
27-May-17  04:11 PM    <DIR>          is-typedarray
27-May-17  04:11 PM    <DIR>          isarray
27-May-17  04:11 PM    <DIR>          isstream
27-May-17  04:11 PM    <DIR>          jodid25519
27-May-17  04:11 PM    <DIR>          jsbn
27-May-17  04:11 PM    <DIR>          json-schema
27-May-17  04:11 PM    <DIR>          json-stable-stringify
27-May-17  04:11 PM    <DIR>          json-stringify-safe
27-May-17  04:11 PM    <DIR>          jsonify
27-May-17  04:11 PM    <DIR>          jsonpointer
27-May-17  04:12 PM    <DIR>          jsprim
27-May-17  04:12 PM    <DIR>          libxmljs
27-May-17  04:11 PM    <DIR>          lodash
27-May-17  04:11 PM    <DIR>          lolex
27-May-17  04:11 PM    <DIR>          long
27-May-17  04:11 PM    <DIR>          lowercase-keys
27-May-17  04:11 PM    <DIR>          mime-db
27-May-17  04:11 PM    <DIR>          mime-types
27-May-17  04:11 PM    <DIR>          minimatch
27-May-17  04:11 PM    <DIR>          minimist
27-May-17  04:11 PM    <DIR>          mkdirp
27-May-17  04:11 PM    <DIR>          ms
27-May-17  04:11 PM    <DIR>          mute-stream
27-May-17  04:11 PM    <DIR>          nan
27-May-17  04:11 PM    <DIR>          ncp
27-May-17  04:11 PM    <DIR>          nock
27-May-17  04:12 PM    <DIR>          node-pre-gyp
27-May-17  04:11 PM    <DIR>          node-status-codes
27-May-17  04:11 PM    <DIR>          node-uuid
27-May-17  04:11 PM    <DIR>          nopt
27-May-17  04:11 PM    <DIR>          npmlog
27-May-17  04:11 PM    <DIR>          number-is-nan
27-May-17  04:11 PM    <DIR>          oauth-sign
27-May-17  04:11 PM    <DIR>          object-assign
27-May-17  04:11 PM    <DIR>          once
27-May-17  04:11 PM    <DIR>          os-homedir
27-May-17  04:11 PM    <DIR>          os-tmpdir
27-May-17  04:11 PM    <DIR>          osenv
27-May-17  04:11 PM    <DIR>          parse-json
27-May-17  04:11 PM    <DIR>          path-is-absolute
27-May-17  04:11 PM    <DIR>          performance-now
27-May-17  04:11 PM    <DIR>          pinkie
27-May-17  04:11 PM    <DIR>          pinkie-promise
27-May-17  04:11 PM    <DIR>          pkginfo
27-May-17  04:11 PM    <DIR>          prepend-http
27-May-17  04:11 PM    <DIR>          process-nextick-args
27-May-17  04:12 PM    <DIR>          prompt
27-May-17  04:11 PM    <DIR>          propagate
27-May-17  04:11 PM    <DIR>          protobufjs
27-May-17  04:11 PM    <DIR>          punycode
27-May-17  04:11 PM    <DIR>          qs
27-May-17  04:12 PM    <DIR>          rc
27-May-17  04:11 PM    <DIR>          read
27-May-17  04:11 PM    <DIR>          read-all-stream
27-May-17  04:11 PM    <DIR>          readable-stream
27-May-17  04:11 PM    <DIR>          request
27-May-17  04:11 PM    <DIR>          revalidator
27-May-17  04:11 PM    <DIR>          rimraf
27-May-17  04:11 PM    <DIR>          safe-buffer
27-May-17  04:11 PM    <DIR>          samsam
27-May-17  04:11 PM    <DIR>          semver
27-May-17  04:11 PM    <DIR>          set-blocking
27-May-17  04:11 PM    <DIR>          signal-exit
27-May-17  04:11 PM    <DIR>          sinon
27-May-17  04:11 PM    <DIR>          sntp
27-May-17  04:12 PM    <DIR>          sshpk
27-May-17  04:11 PM    <DIR>          stack-trace
27-May-17  04:11 PM    <DIR>          steam
27-May-17  04:11 PM    <DIR>          string-width
27-May-17  04:11 PM    <DIR>          stringstream
27-May-17  04:11 PM    <DIR>          string_decoder
27-May-17  04:11 PM    <DIR>          strip-ansi
27-May-17  04:11 PM    <DIR>          strip-json-comments
27-May-17  04:11 PM    <DIR>          supports-color
27-May-17  04:11 PM    <DIR>          tar
27-May-17  04:12 PM    <DIR>          tar-pack
27-May-17  04:11 PM    <DIR>          timed-out
27-May-17  04:11 PM    <DIR>          tough-cookie
27-May-17  04:11 PM    <DIR>          tumblr.js
27-May-17  04:11 PM    <DIR>          tunnel-agent
27-May-17  04:11 PM    <DIR>          tweetnacl
27-May-17  04:11 PM    <DIR>          type-detect
27-May-17  04:11 PM    <DIR>          uid-number
27-May-17  04:11 PM    <DIR>          underscore
27-May-17  04:11 PM    <DIR>          unzip-response
27-May-17  04:11 PM    <DIR>          url-parse-lax
27-May-17  04:12 PM    <DIR>          util
27-May-17  04:11 PM    <DIR>          util-deprecate
27-May-17  04:12 PM    <DIR>          utile
27-May-17  04:11 PM    <DIR>          verror
27-May-17  04:11 PM    <DIR>          wide-align
27-May-17  04:12 PM    <DIR>          winston
27-May-17  04:11 PM    <DIR>          wolfram
27-May-17  04:11 PM    <DIR>          wrappy
27-May-17  04:11 PM    <DIR>          xtend
27-May-17  04:12 PM    <DIR>          youtube-node
               0 File(s)              0 bytes
             191 Dir(s)  56,288,817,152 bytes free

C:\Users\CezmiShef\a\t\node_modules>node example.js
module.js:471
    throw err;
    ^

Error: Cannot find module 'C:\Users\CezmiShef\a\t\node_modules\example.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3`
```

**Comment by bonnici** [comment]
Yeah this is pretty out of date. The example is trying to install the bot code from npm but I handed that over a long time ago. I tried updating the example to use the local source but then I ran into an error in one of the dependencies because everything is out of date by at least one major version. I'll update the example and add a shrinkwrap but even then I'm guessing it will only work on linux unless you want to try to wrangle node-gyp into working. I'd recommend that you try using [Efreak's fork](https://github.com/Steam-Chat-Bot/node-steam-chat-bot) instead of this one.
